### PR TITLE
Static Lich Models + Kel'Thuzad Model

### PR DESCRIPTION
**Pull request by ercarp**
ℹ️ **Development Notice**: Please note that this branch is based on [1-8-1-update](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/tree/1-8-1-update), which means it **_should not_** be merged into `master` until after the `1-8-1-update` branch has been merged.

## Changelog:
- Added **4** static model variants for **_Lich_** creature trait with **4** color variants each. _(The fourth variant is only used by Lich creatures with `tier_duchy` or higher)._
- Added **1** static model for **Kel'Thuzad**. This has a portrait modifier that activates it when he has the **_Lich_** creature trait.
  - Added an exception to **Kel'Thuzad**'s scripted beards and hairstyles so that they are not used when he is a **_Lich_** (this caused a visual bug).

<details><summary>Images</summary>

**Character Finder**
image

**Model Variant 1**
image

**Model Variant 2**
image

**Model Variant 3**
image

**Model Variant 4 (Dukes and above)**
image

**Kel'Thuzad**
image

</details>

## ⚠ Known issues
- [ ] 1. If a lower-ranking **_Lich_** becomes a Duke and their model changes, their color changes at the same time. _(Not a big deal, but could be immersion-breaking for some)._

<details><summary>Images</summary>

image
image

</details>

## Tests:
- [ ] There are no errors in `wc` files in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log` except `portrait_decals.cpp:101`
- [ ] The mod takes less than 5.5 GB in the Task Manager (Windows)

# How to test:
1. Start up **Crusader Kings III** in version `1.8.2` of the game and look for characters with the **_Lich_** creature trait in the 605 bookmark.
2. Provide feedback on this addition. Some things to consider are the placement and size of the model in the character view. Is it too high, too low, too big, too small, etc.

### CODE REVIEW

**Review by CantTouchThisJinTao — APPROVED** [review]
LGTM

**Comment by ElMariuso** [comment]
Looks good to me, nothing to say more

**Review by ElMariuso — APPROVED** [review]
LGTM, nothing to say more

File: common/scripted_triggers/wc_portrait_triggers.txt [hunk 1]
```diff
@@ -244,6 +245,10 @@ has_yaungol_portrait_trigger = {
 	exists = culture
 	is_race_no_gene_trigger = { RACE = creature_yaungol }
 }
+has_yaungol_portrait_trigger = {
+	exists = culture
+	is_race_no_gene_trigger = { RACE = creature_lich }
+}
```
Comment (zumbak04) [inline common/scripted_triggers/wc_portrait_triggers.txt:251]: Duplicate.

File: gfx/portraits/portrait_modifiers/50_beards_scripted_characters.txt [hunk 2]
```diff
@@ -589,6 +589,7 @@ beards_scripted_characters = {
             modifier = {
                 add = 200
 				is_kelthuzad_trigger = yes
+                NOT = { is_race_no_gene_trigger = { RACE = creature_lich }
```
Comment (zumbak04) [inline gfx/portraits/portrait_modifiers/50_beards_scripted_characters.txt:592]: Add to all beard and hair modifiers:
```
modifier = {
	factor = 0
	has_no_portrait_trigger = yes
}
```
Anyone can become a lich at some point I suppose and they will have hair and beard too.

File: common/genes/wc_genes_static_models.txt [hunk 3]
```diff
@@ -1305,6 +1305,47 @@
...
+			boy = male
+			girl = male
+		}
+		static_lich_duke = {
+			index = 88
+			male = {
+				1 = lich4gold
+				1 = lich4bronze
+				1 = lich4silver
+				1 = lich4pewter
```
Comment (zumbak04) [inline common/genes/wc_genes_static_models.txt:1334]: You fix a problem you stated...
Place all golds near each other, then bronze, then silver, then pewter...
```
1 = lich1gold
1 = lich2gold
1 = lich3gold
1 = lich1bronze
etc
```

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: gfx/portraits/portrait_modifiers/50_beards_scripted_characters.txt (hunk 2 above)
Comment (ercarp) [reply gfx/portraits/portrait_modifiers/50_beards_scripted_characters.txt]: This didn't seem to work for some reason so I changed the code a little bit. Added to all hairstyles and beards in b39fff224466497201cf1050022d51a5fa8a5e4d.
```
modifier = {
	add = -1000
	has_no_portrait_trigger = yes
}
```

File: common/genes/wc_genes_static_models.txt (hunk 3 above)
Comment (ercarp) [reply common/genes/wc_genes_static_models.txt]: Applied change in b39fff224466497201cf1050022d51a5fa8a5e4d. This makes it a bit more consistent, but doesn't fix the problem entirely. Bronze Lich models in particular seem to almost always become pewter-colored when they become Dukes.

File: common/scripted_triggers/wc_portrait_triggers.txt (hunk 1 above)
Comment (ercarp) [reply common/scripted_triggers/wc_portrait_triggers.txt]: This was an oversight, I renamed it to `has_lich_portrait_trigger` in b39fff224466497201cf1050022d51a5fa8a5e4d.

File: gfx/portraits/portrait_modifiers/50_beards_scripted_characters.txt (hunk 2 above)
Comment (ercarp) [reply gfx/portraits/portrait_modifiers/50_beards_scripted_characters.txt]: Fixed in b0ebba5c4f4c3248c05cf2b32ef9f288cb3dc5cb.

**Comment by zumbak04** [comment]
> If a lower-ranking Lich becomes a Duke and their model changes, their color changes at the same time. (Not a big deal, but could be immersion-breaking for some).

Got what is wrong. When the portrait modifier is applied it's always a random number. Most likely there's no way if we keep using the current way to apply models.

**Review by zumbak04 — APPROVED** [review]
LGTM!

**Review by TheDrPibb1999 — APPROVED** [review]
LGTM!!

**Pull request merged**